GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift

```diff
@@ -136,3 +137,84 @@ class GameEngine {
         removeDelegate.removeEnemyReachedLine(enemyEntity)
     }
 }
+
+/** Extension for the Game Engine to deal with Mana-related events and logic. */
+extension GameEngine: DroppedManaResponderType {
```

Comment:
I feel like we should move all these logic into the mana system, and keep the game engine clean and just act as a notification center, and interacting with the game meta data to affect the notification calls, thoughts everyone? @brianyenna @andyylam @JermyTan

Reply:
Yup, agreed. 

So basically, the `GameEngine` class will have all these functions like `removeMonster`, `tapDroppedMana` etc., but these functions simply call the relevant functions in the component systems (so `tapDroppedMana` in `ManaSystem`)? Is this what you mean?

Reply:
Agree, basically the game engine will coordinate the action by calling appropriate delegates (component systems) and at the same time can do any pre/post-processing of the result. Is it?